instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/EvReroute.kt

```diff
@@ -0,0 +1,219 @@
+            RouteProgressState.OFF_ROUTE
+        )
+    }
+
+    override fun setupMockLocation(): Location {
+        val mockRoute = RoutesProvider.california_ev_route(activity)
+        return mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = mockRoute.routeWaypoints.first().latitude()
+            longitude = mockRoute.routeWaypoints.first().longitude()
+        }
+    }
+
+    @Test
+    fun ev_reroute() {
+        val mockRoute = RoutesProvider.california_ev_route(activity)
+        val originLocation = mockRoute.routeWaypoints.first()
+        val offRouteLocationUpdate = mockLocationUpdatesRule.generateLocationUpdate {
+            latitude = originLocation.latitude() + 0.005
+            longitude = originLocation.longitude()
+        }
+
+        mockWebServerRule.requestHandlers.addAll(mockRoute.mockRequestHandlers)
+        mockWebServerRule.requestHandlers.add(
+            MockDirectionsRequestHandler(
+                profile = DirectionsCriteria.PROFILE_DRIVING_TRAFFIC,
+                jsonResponse = readRawFileText(
+                    activity,
+                    R.raw.ev_route_response_reroute_ca_2_ev_waypoints
+                ),
+                expectedCoordinates = listOf(
+                    Point.fromLngLat(
+                        offRouteLocationUpdate.longitude,
+                        offRouteLocationUpdate.latitude
+                    ),
+                    mockRoute.routeWaypoints.last()
+                ),
+                relaxedExpectedCoordinates = true
+            )
+        )
+        locationTrackingIdlingResource.register()
+
+        val expectedStates = RouteProgressStateTransitionAssertion(mapboxNavigation) {
+            requiredState(RouteProgressState.TRACKING)
+            requiredState(RouteProgressState.OFF_ROUTE)
+            optionalState(RouteProgressState.INITIALIZED)
+            requiredState(RouteProgressState.TRACKING)
+        }
+
+        // start a route
+        runOnMainSync {
+            mapboxNavigation.historyRecorder.startRecording()
+            mapboxNavigation.startTripSession()
+            mapboxNavigation.requestRoutes(
+                RouteOptions.builder()
+                    .applyDefaultNavigationOptions()
+                    .applyLanguageAndVoiceUnitOptions(activity)
+                    .baseUrl(mockWebServerRule.baseUrl)
+                    .coordinatesList(mockRoute.routeWaypoints)
+                    .unrecognizedProperties(
+                        mapOf(
+                            "engine" to "electric",
+                            "ev_max_charge" to "8000",
+                            "ev_connector_types" to "tesla",
+                            "energy_consumption_curve" to "0,300;20,160;80,140;120,180",
+                            "ev_charging_curve" to "40000,70000",
+                        )
+                    )
+                    .build(),
+                object : NavigationRouterCallback {
```

Comment:
Instead of creating a callback you can do:
```
val routes = mapboxNavigation.requestRoutes(options)
            .getSuccessfulResultOrThrowException()
            .routes
```

Reply:
I wouldn't change what is actually working

Reply:
That's 3 lines of code instead of 20. The tests are already large enough. 
The suspend version of this method does exactly what you do. It's your call of course but I don't see any reason why not.

Reply:
Actually, not only 3: requires to refactor every `runOnMainSync` as well

Reply:
You can just use `sdkTest {}` for the whole test: it's also blocking and on the main thread. Also minus several lines.

Reply:
so, it's already not 3 lines 😃 let's change it if requires any other changes as it just adds a bit of sugar code